js/CornerWidgetUI.js

```diff
@@ -743,64 +812,96 @@ CornerWidgetUI._raise_lead = function (btn_ref){
+		}
+	} else {
+		// if tel input validation isn't working then use 
+		if (CornerWidgetUI.jQuery("#"+CornerWidgetUI.constants._uivar_leadtelID).val().length < 4) {
+			CornerWidgetUI._fail_submit(CornerWidgetUI.constants._uivar_leadtelID)
+			$lucep.send_intelligence(
+				{
+					event_type: "bad-number-entry",
+					payload: {
+						number: tel_no,
+						validationCount: CornerWidgetUI._ui_config.validation.count
+					}
+				}
+			)
+			return false;
+		}
+	}
+
+	// Validate name field
+	if (document["getElementById"](CornerWidgetUI.constants._uivar_leadnameID)["value"]["length"] < 2) {
+		CornerWidgetUI._fail_submit(CornerWidgetUI.constants._uivar_leadnameID)
+		$lucep.send_intelligence(
+			{
+				event_type: "invalid-name-entry",
+				payload: {
+					name: document["getElementById"](CornerWidgetUI.constants._uivar_leadserviceID)["value"],
+					validationCount: CornerWidgetUI._ui_config.validation.count
+				}
+			}
+		)
 		return false;
 	}
+
+
+	lead_name = document["getElementById"](CornerWidgetUI.constants._uivar_leadserviceID)["value"];
 	CornerWidgetUI._ui_config.validation.count = 0; //reset the validation restriction as the criteria passed
 	
 	//Store data for future use
 	$lucep["put_data"]( {"key": "name", "value": document["getElementById"](CornerWidgetUI.constants._uivar_leadnameID)["value"] } );
 	$lucep["put_data"]( {"key": "tel", "value": tel_no } );
 	
-	//trigger lead fire
-	$lucep["raise_lead"](
-		{"payload": {},
-		 "service_id": document["getElementById"](CornerWidgetUI.constants._uivar_leadserviceID)["value"],
-		 "service_name": window["jQuery"]("#gorillaFormService option:selected")["text"](),
-		 "phone_number": tel_no,
-		 "name": document["getElementById"](CornerWidgetUI.constants._uivar_leadnameID)["value"],
-		 "callback": function (progress){
-			 if (!progress["ticket-status"] && progress["server"]===true){
-				 //there was an error on the server, invite the user to try again
-				 //TODO: Invite the user to try again
-				 CornerWidgetUI.control({"state": "new"});
-				 return;
-			 }
-			 //gets notified on the various stages of lead progression/notification
-			 switch(progress["ticket-status"]){
-			 case "new":
-				 CornerWidgetUI.control({"state": "raised_lead"});
-				 return true; // more updates wanted
-				 break;
-				 
-			 case "processing":
-				 CornerWidgetUI.control({"state": "in_call"});
-				 return true; //more updates wanted
-				 break;
-				 
-			 case "finished":
-				 CornerWidgetUI.control({"state": "new"});
-				 return false; //no more updates wanted on this ticket
-				 break;
-				 
-			 case "no-ticket":
-				 CornerWidgetUI.control({"state": "new"});
-				 return false; //no more updates wanted on this ticket
-				 break;
-				 
-			 }
-		 }
+	var lead_opts = {
+		"payload": {},
+		"service_id": document["getElementById"](CornerWidgetUI.constants._uivar_leadserviceID)["value"],
+		"service_name": CornerWidgetUI.jQuery("#gorillaFormService option:selected")["text"](),
```

Comment:
We already have jQuery, which we know at this point has definitely loaded for sure even on slow network, might as well use it right? Even in the current version you get service_name using jQuery